test/partiql-tests-runner/src/test/kotlin/org/partiql/runner/io/DatumIonReader.kt

```diff
@@ -291,6 +298,65 @@ class DatumIonReader(
+                                else -> throw IllegalArgumentException("unexpected field: ${reader.fieldName}")
+                            }
+                        }
+                        reader.stepOut()
+                        val signMultiplier = if (sign == "-") -1 else 1
+                        when {
+                            years < 0 -> throw IllegalArgumentException("years must be non-negative, got: $years")
+                            months < 0 -> throw IllegalArgumentException("months must be non-negative, got: $months")
+                        }
+                        Datum.intervalYearMonth(signMultiplier * years, signMultiplier * months, INTERVAL_MAX_PRECISION)
+                    }
+                    PARTIQL_ANNOTATION.INTERVAL_DT_ANNOTATION -> {
+                        reader.stepIn()
+                        var sign: String? = null
+                        var days = 0
+                        var hours = 0
+                        var minutes = 0
+                        var seconds = 0
+                        var nanos = 0
+                        while (reader.next() != null) {
+                            when (reader.fieldName) {
+                                "sign" -> sign = reader.stringValue()
+                                "days" -> days = reader.intValue()
+                                "hours" -> hours = reader.intValue()
+                                "minutes" -> minutes = reader.intValue()
+                                "seconds" -> seconds = reader.intValue()
+                                "nanos" -> nanos = reader.intValue()
+                                else -> throw IllegalArgumentException("unexpected field: ${reader.fieldName}")
+                            }
+                        }
+                        reader.stepOut()
+                        val signMultiplier = if (sign == "-") -1 else 1
+                        when {
+                            days < 0 -> throw IllegalArgumentException("days must be non-negative, got: $days")
+                            hours < 0 -> throw IllegalArgumentException("hours must be non-negative, got: $hours")
+                            minutes < 0 -> throw IllegalArgumentException("minutes must be non-negative, got: $minutes")
+                            seconds < 0 -> throw IllegalArgumentException("seconds must be non-negative, got: $seconds")
+                            nanos < 0 -> throw IllegalArgumentException("nanos must be non-negative, got: $nanos")
+                        }
+                        Datum.intervalDaySecond(
+                            signMultiplier * days,
+                            signMultiplier * hours,
+                            signMultiplier * minutes,
+                            signMultiplier * seconds,
+                            signMultiplier * nanos,
+                            INTERVAL_MAX_PRECISION,
+                            INTERVAL_MAX_PRECISION,
```

Comment:
nit: even though the max nano precision and max interval leading field precision is the same value, it's probably more accurate to have this as its own variable